app/controllers/trips_controller.rb

```diff
@@ -1,67 +1,53 @@
 class TripsController < ApplicationController
 
-  # user is NOT logged in
-    # remove/filter out private trips
-    #  user_signed_in? -> from devise
-  # user IS logged in
-    # Dash
-      # only public
-    # your trips
-      # specific current user's publics and privates
-    def index 
-        trips = Trip.all
-        render :json => trips.to_json(include: :sights)
+  def index 
+    trips = Trip.all
+    if user_signed_in?
+      filtered_trips = trips.where("user_id = #{current_user.id} or is_public = 'True'")
+      render :json => filtered_trips.to_json(include: :sights)
+    else
+      public_trips = trips.where("is_public = 'True'")
+      render :json => public_trips.to_json(include: :sights)
```

Comment:
```suggestion
    query_string = "is_public = 'True' #{user_signed_in? ? ' or user_id = #{current_user.id}' : ''}"

      filtered_trips = trips.where(query_string)
      render :json => filtered_trips.to_json(include: :sights)
```

to reduce redundant code, you can built the query string and then pass it to the `where` method since both the if and else blocks have the same instructions